### DISCUSSION TITLE
[FEATURE REQUEST] add ability for ini.options_present to hide output using show_changes and output_loglevel

### DISCUSSION THREAD

**Discussion by alias454** [saltstack/salt#67327] [Feature Requests]
When using the **ini.options_present** state, I cannot pass sensitive data directly in the state. I currently have to set a value that can later be replaced using file.replace, which has the ability to hide output using show_changes and output_loglevel.

**Describe the solution you'd like**
Gain an ability to hide sensitive information from logging output and restrict output returned in the run results like the file module and other modules that already implement these features. 

Currently, I have a workaround where I create the ini structure using place holders like REPLACE_ME. I then use file.replace to match the string to replace with the actual value needed on the config file but this adds several additional tasks where it would be nice if the ini module had the ability to redact information natively. 

**Additional context**
```
----------
          ID: ini-add-sslConfig-section
    Function: ini.options_present
        Name: /opt/dev/server.conf
      Result: True
     Comment: Changes take effect
     Started: 23:04:47.224928
    Duration: 37.952 ms
     Changes:   
              ----------
              sslConfig:
                  ----------
                  sslPassword:
                      ----------
                      after:
                          REPLACE_ME
                      before:
                          None
----------
          ID: comment-sslconfig-sslpassword-value
    Function: file.replace
        Name: /opt/dev/server.conf
      Result: True
     Comment: Changes were made
     Started: 23:04:47.279006
    Duration: 9.481 ms
     Changes:   
              ----------
              diff:
                  True
```

Thanks

**Comment by OrangeDog** [comment]
I think in general any state could leak sensitive data via its changes, which is why there's a global don't show changes option, and the logs are restricted to `root` by default.

I'm not sure it's a good idea to add specific parameters to individual states to hide their changes. A global argument perhaps (like `reload_grains` or `runas`) would be a better way to do it, if it's actually useful to do.

**Comment by alias454** [comment]
The global setting is great but may not be a reliable mechanism when sharing formulas with other people. Logs being owned by root is another decent default but sometimes those logs are shipped to a logging platform. 

I certainly won't say I know the correct way to implement this as a feature but if it makes the cut, having a consistent global option seems to make sense. How are those features implemented with the file state?

Thanks
